Update report generator to use reportlab

DJensen94 opened the pull request:
Fixes #420 
## 🗣 Description ##

Our current report generator uses an html template and then converts the html into a pdf report. The library that does the transformation does not have enough features to implement the fonts and dynamic spacing our reports require. This means we will have to redesign the report generation process using a new library.

## Motivation ##

This is change is required to have our reports meet the CISA style guide and also make our reports more robust as we have had issues where complete charts have vanished due to sizing issues.

## 🧪 Testing ##

Reports are correctly generated without any missing visuals 
Fonts are correctly displayed

## ✅ Pre-approval checklist ##

- [X] This PR has an informative and human-readable title.
- [X] Changes are limited to a single goal - *eschew scope creep!*
- [X] *All* future TODOs are captured in issues, which are referenced in code comments.
- [X] All relevant type-of-change labels have been added.
- [X] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [X] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [X] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.

DJensen94 committed:
Update report generator to use reportlab

update generator to use reportlab to make report compliant with CISA standards

DJensen94 committed:
remove commented out variable call

remove unused commented out variable

DJensen94 committed:
add demoji dependency

add missing dependency demoji

DJensen94 committed:
add fonts

add fonts for the report generator to use

DJensen94 committed:
update tests

update report generator test

DJensen94 committed:
update cred tests

update test credential metrics test

DJensen94 committed:
update report gen test

updated report gen to have correct data dictionary structure

DJensen94 committed:
generate image in test

generate testing images for report generator test

DJensen94 committed:
Update path of image calls

update path of image calls

DJensen94 committed:
update file that is replaced when testing

update image used to replace missing images when testing

DJensen94 committed:
Fix table column names in test

in report gen test fix the column names in test dict

DJensen94 committed:
Add additional images to test

add required images to testing

DJensen94 committed:
add output filename to mock

output filename from the embed mock

DJensen94 committed:
add return statement

add return statement to generate reports

DJensen94 committed:
Correctly capitalize comments

Ensure all comments start with a capital letter

jsf9k commented on src/pe_reports/data/db_query.py:
```diff
@@ -138,12 +138,36 @@ def query_breachdetails_view(org_uid, start_date, end_date):
 
 def query_domMasq(org_uid, start_date, end_date):
-    """Query domain masquerading table."""
+    """Query domain_permuations associated witha given org."""
```
```suggestion
    """Query domain_permuations associated with a given org."""
```

jsf9k commented on src/pe_reports/report_generator.py:
```diff
@@ -41,42 +41,64 @@ def embed(
     mi_xlsx,
 ):
-    """Embeds raw data into PDF and encrypts file."""
+    """Embed raw data into PDF and encrypts file."""
```
```suggestion
    """Embed raw data into PDF and encrypt file."""
```

jsf9k commented on src/pe_reports/reportlab_generator.py:
```suggestion
                            positive findings; however, these findings should be treated as information that your
```

jsf9k commented on src/pe_reports/reportlab_generator.py:
```suggestion
                    <font face="Franklin_Gothic_Medium_Regular">Table 3</font> shows registered or updated domains that were
```

jsf9k commented on src/pe_reports/reportlab_generator.py:
```suggestion
                    assets such as domain names and IPs.
```

jsf9k commented on src/pe_reports/reportlab_generator.py:
```suggestion
                about CVEs can be found <link href="https://nvd.nist.gov/">here</link>.""",
```

jsf9k commented on src/pe_reports/reportlab_generator.py:
```suggestion
            network compromise, or misconfiguration of databases leading to public exposure. This leaked data is then listed
```

jsf9k commented on src/pe_reports/reportlab_generator.py:
```suggestion
                    will often be published or offered 'for sale'. This information along with a list of the most active CVEs on the
```

jsf9k reviewed (commented):
This looks OK to my uneducated eyes.  I noticed a few typos and such, so do with those what you will.

DJensen94 committed:
Fix typos in docstring

Fix typo in docstring

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

DJensen94 committed:
Fix typo in docstring

fix typo of "encrypts" to "encrypt' in docstring

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

DJensen94 committed:
replace comma with semi colon

update FAQ to have correct punctuation

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

DJensen94 committed:
Remove extra space

Remove extra space in paragraph

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

DJensen94 committed:
remove unnecessary apostrophe

Remove unnecessary apostrophe on "IP's" to "IPs"

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

DJensen94 committed:
remove unnecessary apostrophe

Remove unnecessary apostrophe in "CVE's"

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

DJensen94 committed:
add an oxford comma, where necessary

Add comma to end of list of services

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

DJensen94 committed:
remove incorrect apostrophe from "CVE's"

remove incorrect apostrophe from "CVE's" to "CVE"

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

DJensen94 committed:
Merge branch 'develop' into DJ_Reportlab_generator_PR

DJensen94 committed:
Ran precommits to fix lint check

DJensen94 commented:
@jsf9k after merging Develop into this branch to get them in sync, the ansible lint check started failing, I don't think that is an issue on our end. Any ideas on that issue?

jsf9k commented:
> @jsf9k after merging Develop into this branch to get them in sync, the ansible lint check started failing, I don't think that is an issue on our end. Any ideas on that issue?

This failure is due to some upstream changes.  The next set of Lineage changes that trickle down should fix the problem.

DJensen94 committed:
Create core report functionality

update report generator to create core reports

DJensen94 committed:
Merge branch 'develop' into DJ_Reportlab_generator_PR

DJensen94 committed:
update test to work with premium reports

update test to verify it is a premium report before testing generating

dav3r commented on src/pe_reports/charts.py:
```diff
@@ -161,43 +165,123 @@ def line_chart(self):
+                    y1_offset = 8
+                    y2_offset = 8
+                print(y1)
+                print(y2)
```
Is there any value in printing these out?  Or should they be logged as debug statements?

DJensen94 replied:
These were unnecessary print statements used during development testing, they were removed in this commit: [349eb86](https://github.com/cisagov/pe-reports/pull/565/commits/349eb86247c47bebd0cc468cd8f7dcf258b53d4b)

dav3r commented on src/pe_reports/pe_reports_django_project/.env:
I'm assuming this empty file doesn't need to be checked in to the repo?

DJensen94 replied:
for some reason the commit wouldn't pass checks without the .env file

dav3r replied:
What was the error message that you got?

DJensen94 replied:
I don't remember off the top of my head, it wouldn't let it through the checks, I can try taking it out though and seeing what comes up.

DJensen94 replied:
<img width="613" alt="image" src="https://github.com/cisagov/pe-reports/assets/79864006/eb1bb650-9deb-4ea9-b974-2d5fbf02cc07">

Here is the recreation of the commit error

dav3r replied:
Can you please let us know which pre-commit check was throwing that `Couldn't find env file` error?  Is it still happening?

DJensen94 replied:
I re-deleted, and this time it passed checks in this commit: [8f88e57](https://github.com/cisagov/pe-reports/pull/565/commits/8f88e576025e53d1b075b5dfc986dc912952768d)

dav3r commented on src/pe_reports/report_generator.py:
```diff
@@ -11,6 +11,7 @@
   -l --log-level=LEVEL              If specified, then the log level will be set to
                                     the specified value.  Valid values are "debug", "info",
                                     "warning", "error", and "critical". [default: info]
+  -sc --soc_med_included            Include social media posts from Cybersixgill in the report.
```
Typically a short CLI flag is just a single character, i.e. `-s` instead of `-sc`.

DJensen94 replied:
Issue addressed in the following commit: [ec0f8be](https://github.com/cisagov/pe-reports/pull/565/commits/ec0f8bea0e02715129c7e6e197e01b7e07b00682)

dav3r commented on src/pe_reports/report_generator.py:
```diff
@@ -11,6 +11,7 @@
   -l --log-level=LEVEL              If specified, then the log level will be set to
                                     the specified value.  Valid values are "debug", "info",
                                     "warning", "error", and "critical". [default: info]
+  -sc --soc_med_included            Include social media posts from Cybersixgill in the report.
```
Also, you need to add this optional flag to the ["Usage" line above](https://github.com/cisagov/pe-reports/blob/18e3d1c69fccc85afc0506ed7382e3691b9c5c3d/src/pe_reports/report_generator.py#L4) so that docopt knows about it:
```
pe-reports REPORT_DATE OUTPUT_DIRECTORY [--log-level=LEVEL] [--soc_med_included]
```

DJensen94 replied:
Issue addressed in the following commit: [ec0f8be](https://github.com/cisagov/pe-reports/pull/565/commits/ec0f8bea0e02715129c7e6e197e01b7e07b00682)

dav3r commented on src/pe_reports/report_generator.py:
```diff
@@ -264,13 +259,15 @@ def main():
     if not os.path.exists(validated_args["OUTPUT_DIRECTORY"]):
         os.mkdir(validated_args["OUTPUT_DIRECTORY"])
 
+    try:
+        soc_med = validated_args["--soc_med_included"]
+    except Exception as e:
+        LOGGER.info(f"Social media should not included: {e}")
+        soc_med = False
```
If you follow my earlier suggestion related to docopt, you can remove this block and just make the change below.
```suggestion
```

dav3r commented on src/pe_reports/report_generator.py:
```diff
@@ -264,13 +259,15 @@ def main():
-        validated_args["REPORT_DATE"],
-        validated_args["OUTPUT_DIRECTORY"],
+    generate_reports(
+        validated_args["REPORT_DATE"], validated_args["OUTPUT_DIRECTORY"], soc_med
```
Related to suggestion above.  Note, I didn't run this thru `black`, so it might get reformatted due to long line length:
```suggestion
        validated_args["REPORT_DATE"], validated_args["OUTPUT_DIRECTORY"], validated_args["--soc_med_included"]
```

dav3r reviewed (changes requested):
I didn't quite get all the way through my review (2 big files left), but here are some items to work on in the meantime.

jsf9k commented on src/pe_reports/metrics.py:
```diff
@@ -133,7 +137,7 @@ def summary(self):
                 ]
             ]
-            domain_sum = domain_sum[:25]
+            domain_sum.loc[domain_sum["ipv6"] == "", "ipv6"] = "NA"
```
It's not clear to me what is happening in this line.  Can you add a comment that explains?

DJensen94 replied:
Added comment in this commit:[d456379](https://github.com/cisagov/pe-reports/pull/565/commits/d45637968b769748453b6999e457719a96525757)

jsf9k commented on src/pe_reports/pages.py:
```diff
@@ -88,42 +55,76 @@ def credential(chevron_dict, trending_start_date, start_date, end_date, org_uid)
         "pw_creds": Credential.password(),
-        "breach_table": breach_table,
-        "breachAppendix": buildAppendixList(Credential.breach_appendix()),
+        "breach_table": Credential.breach_details()[:15],
```
Is `15` a magic number here, or are you truncating a string for some reason?

DJensen94 replied:
We only want to display the first 15 rows of the dataframe in the report, so this may be considered a magic number. Should I create a variable called table_rows and pass it there?

dav3r replied:
At minimum, I'd recommend adding a comment to that effect.  If you think it could be useful, then replacing it with a variable is also not a bad idea.  @jsf9k may have other thoughts, but I figured I'd chime in here.

jsf9k replied:
I'd argue for naming the variable something more specific and descriptive than `table_rows`.  Please add a variable, and a comment describing what the variable represents.

DJensen94 replied:
Addressed in the following commit: [532ff4d](https://github.com/cisagov/pe-reports/pull/565/commits/532ff4d1f303fdb620238bf983d7b3a3738e25b7)

jsf9k reviewed (commented):
I had a few questions.

DJensen94 committed:
Remove testing print statements in charts.py

remove print statements in charts.py used for testing

DJensen94 committed:
Add social media flag to docopt usage line

add social media flag to the docopt usage line.

DJensen94 committed:
Remove unnecessary validation code

Since we added the social media flag to the usage line docopt will validate the entry and we don't need to have our own validation

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Add docopt flag directly to function call

Since docopt is validating thee social media flag we can send it straight to the function call

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
add comment describing how to update empty IPv6 values

add comment describing how to update empty IPv6 values

DJensen94 committed:
Merge branch 'DJ_Reportlab_generator_PR' of https://github.com/cisagov/pe-reports into DJ_Reportlab_generator_PR

DJensen94 committed:
Put function parameters on there own line

Put function parameters on there own line so the line isn't too long to pass lint checks

DJensen94 committed:
test .env delete

test removal of .env deletion

DJensen94 committed:
Remove magic number from pages

Remove magic number setting how many rows show up in the breach table.

dav3r commented:
@DJensen94 While you are waiting for the remaining reviews, you can merge the latest changes from `develop` into this branch to ensure there are no issues there?

DJensen94 committed:
Merge branch 'develop' into DJ_Reportlab_generator_PR
Merge changes in develop branch into reportlab branch

dav3r commented on src/pe_reports/report_generator.py:
```diff
@@ -11,6 +11,7 @@
   -l --log-level=LEVEL              If specified, then the log level will be set to
                                     the specified value.  Valid values are "debug", "info",
                                     "warning", "error", and "critical". [default: info]
+  -s --soc_med_included            Include social media posts from Cybersixgill in the report.
```
Fix spacing in usage text to maintain alignment (it got messed up when the flag was changed from `-sc` to `-s`).
```suggestion
  -s --soc_med_included             Include social media posts from Cybersixgill in the report.
```

dav3r commented on src/pe_reports/report_generator.py:
```diff
@@ -141,70 +152,53 @@ def generate_reports(datestring, output_directory):
-            top_cves.to_excel(miWriter, sheet_name="Top CVEs", index=False)
-            miWriter.save()
+            output_filename = f"{output_directory}/Posture_and_Exposure_Report-{org_code}-{datestring}.pdf"
+            # convert_html_to_pdf(source_html, output_filename)#TODO possibly generate report here
```
Clean up comment formatting.  Also, there should be a corresponding GitHub issue for this TODO and the issue URL should be included in the comment here.
```suggestion
            # TODO possibly generate report here
            # convert_html_to_pdf(source_html, output_filename)
```

DJensen94 replied:
Removed commented out code and unnecessary TODO statement in the following commit: [95dd8b4](https://github.com/cisagov/pe-reports/pull/565/commits/95dd8b4f7fa2cd11df67f071bc73892e41edec80)

dav3r commented on src/pe_reports/report_generator.py:
```diff
@@ -141,70 +152,53 @@ def generate_reports(datestring, output_directory):
+            pdf = f"{output_directory}/Posture_and_Exposure_Report-{org_code}-{datestring}.pdf"
 
-            (filesize, tooLarge) = embed(
+            # Embed excel and Json files
```
Use standard capitalization:
```suggestion
            # Embed Excel and JSON files
```

dav3r commented on src/pe_reports/reportlab_generator.py:
```diff
@@ -0,0 +1,1839 @@
+        "Franklin_Gothic_Medium_Regular",
+        BASE_DIR + "/fonts/FranklinGothicMediumRegular.ttf",
+    )
+)
```
Replace with a loop (I think this will pass linting, but I didn't check):
```suggestion
for font_name, font_filename in [
    ("Franklin_Gothic_Book", "FranklinGothicBook.ttf"),
    ("Franklin_Gothic_Book_Italic", "FranklinGothicBookItalic.ttf"),
    ("Franklin_Gothic_Demi_Regular", "FranklinGothicDemiRegular.ttf"),
    ("Franklin_Gothic_Medium_Italic", "FranklinGothicMediumItalic.ttf"),
    ("Franklin_Gothic_Medium_Regular", "FranklinGothicMediumRegular.ttf"),
    ]:
    pdfmetrics.registerFont(TTFont(font_name, BASE_DIR + "/fonts/" + font_filename))
```

DJensen94 replied:
fixed linting check in this commit: [0579465](https://github.com/cisagov/pe-reports/pull/565/commits/057946512e6e9f5991ec93555f62218f75120f3c)

dav3r commented on src/pe_reports/reportlab_generator.py:
```diff
@@ -0,0 +1,1839 @@
+    return sha256(s.encode("utf-8")).hexdigest()
+
+
+# Extend Table of contents to create a List of Figures Class
```
Clean up comment:
```suggestion
# Extend TableOfContents class to create ListOfFigures class
```

dav3r commented on src/pe_reports/reportlab_generator.py:
```diff
@@ -0,0 +1,1839 @@
+
+# Extend Table of contents to create a List of Figures Class
+class ListOfFigures(TableOfContents):
+    """Class extention to build a Table of Figures."""
```
Simplify comment:
```suggestion
    """Build a table of figures."""
```

dav3r commented on src/pe_reports/reportlab_generator.py:
```diff
@@ -0,0 +1,1839 @@
+
+    def notify(self, kind, stuff):
+        """
+        Call he notification hook to register all kinds of events.
```
Fix typo:
```suggestion
        Call the notification hook to register all kinds of events.
```

dav3r commented on src/pe_reports/reportlab_generator.py:
```diff
@@ -0,0 +1,1839 @@
+            self.addEntry(*stuff)
+
+
+# Extend Table of contents to create a List of Tables Class
```
Clean up comment:
```suggestion
# Extend TableOfContents class to create ListOfTables class
```

dav3r commented on src/pe_reports/reportlab_generator.py:
```diff
@@ -0,0 +1,1839 @@
+
+# Extend Table of contents to create a List of Tables Class
+class ListOfTables(TableOfContents):
+    """Class extention to build a Table of Tables."""
```
Simplify comment:
```suggestion
    """Build a table of tables."""
```

dav3r commented on src/pe_reports/reportlab_generator.py:
```diff
@@ -0,0 +1,1839 @@
+
+
+class MyDocTemplate(BaseDocTemplate):
+    """Extend the BaseDocTemplate to adjust Template."""
```
Simplify comment:
```suggestion
    """Customize the document template."""
```

dav3r commented on src/pe_reports/reportlab_generator.py:
```diff
@@ -0,0 +1,1839 @@
+
+
+def report_gen(data_dict, soc_med_included=False):
+    """Generate a P&E report with data passed in the data dictionry."""
```
Fix typo:
```suggestion
    """Generate a P&E report with data passed in the data dictionary."""
```

dav3r commented on src/pe_reports/reportlab_generator.py:
Clean up comment:
```suggestion
    # Document structures
```

dav3r commented on src/pe_reports/reportlab_generator.py:
```suggestion
```

dav3r commented on src/pe_reports/reportlab_generator.py:
```suggestion
    # Title page
```

dav3r commented on src/pe_reports/reportlab_generator.py:
```suggestion
    # Table of contents
```

dav3r commented on src/pe_reports/reportlab_generator.py:
```suggestion
    # Set styles for levels in table of contents
```

dav3r commented on src/pe_reports/reportlab_generator.py:
```suggestion
    # Table of figures and table of contents
```

dav3r commented on src/pe_reports/reportlab_generator.py:
```suggestion
    # Content pages
    # Start Introduction page
```

dav3r commented on src/pe_reports/reportlab_generator.py:
```suggestion
    # Start generating Summary page
```

dav3r commented on src/pe_reports/reportlab_generator.py:
```suggestion
    # Start generating Credentials page
```

dav3r commented on src/pe_reports/reportlab_generator.py:
Can this line be deleted?
```suggestion
```

dav3r commented on src/pe_reports/reportlab_generator.py:
```suggestion
    # Start generating Domain Masquerading page
```

dav3r commented on src/pe_reports/reportlab_generator.py:
Can this line be deleted?
```suggestion
```

dav3r commented on src/pe_reports/reportlab_generator.py:
```suggestion
    # Start generating Vulnerabilities page
```

dav3r commented on src/pe_reports/reportlab_generator.py:
Can this line be deleted?
```suggestion
```

dav3r commented on src/pe_reports/reportlab_generator.py:
Can this line be deleted?
```suggestion
```

dav3r commented on src/pe_reports/reportlab_generator.py:
```suggestion
    # Start generating Dark Web page
```

dav3r commented on src/pe_reports/reportlab_generator.py:
Can this line be deleted?
```suggestion
```

dav3r commented on src/pe_reports/reportlab_generator.py:
```suggestion
    # Start generating Methodology page
```

dav3r reviewed (changes requested):
Here are a few more mostly cosmetic suggestions.

DJensen94 committed:
Clean up comment

Clean up comment by removing asterisks

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
delete commented out line

remove commented out line that is no longer used

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Clean up comments

Remove unnecessary asterisks from comment

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
remove commented out line

remove unecessary commented out code

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
remove commented out line

Remove commented out code

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Fix spacing in docopt argument definition

Add extra space so docopt descriptions all align

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Remove unnecessary comment

Remove commented out code

DJensen94 committed:
Capitalize filetype names

Capitalize filetype names in comment

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Loop through font imports to remove duplication

Loop through the font imports to simplify the code and remove a majority of the duplicate code

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Clean up comments to maintain class naming

Clean up the comments so they correctly match class names

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
fix formatting on font import

font import is failing linter, need to adjust it to trigger the linter locally

DJensen94 committed:
fix mispelling in docstring

Correct spelling in docstring

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Simplify Comment

Simplify the docstring for ListOfFigures class

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Clean up section comment

Clean up section comment, removing asterisks and updating capitalization

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Clean up comment to reflect class names

Clean up comment to reflect the class names

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Simplify Comment

Simplify comment, don't need to say it is a class

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Simplify Comment

Its not necessary to say which class is extended

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
fix typo

correct misspelling of dictionary

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Clean up comment

remove asterisks and fix capitalization

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Remove Section divider string

Remove line of asterisks

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Clean up comment

remove Asterisks from string

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Clean up comment

remove asterisks from comment

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Fix capitalization

Fix capitalization in comment

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Remove asterisks from comment

Fix capitalization and remove asterisks from comment

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
delete commented out code

delete commented out code

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
clean up comment

remove asterisks and fix capitalization

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Delete commented out code

Delete commented out code

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Clean up comment

Remove asterisk and fix capitalization

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Clean up comment

Remove asterisk and fix capitalization

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

DJensen94 committed:
Clean up comments

remove asterisks and fix capitalization

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

dav3r commented:
@DJensen94 - For future reference, I suggest you use the [batch functionality](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes) to accept and commit multiple **similar/related** suggestions (e.g. https://github.com/cisagov/pe-reports/pull/565/commits/4a4b7562bdaa84300e6c60c123e0ecfd57af00e0, https://github.com/cisagov/pe-reports/pull/565/commits/40f073f6e8457e91f608215b2dde417cc1343061, https://github.com/cisagov/pe-reports/pull/565/commits/e2502bb64051237916f7a878dc2481abf552e37f, etc.)

This would reduce the number of commits you need to make in cases like this.

jsf9k commented on src/pe_reports/report_generator.py:
```diff
@@ -1,7 +1,7 @@
 
 Usage:
-  pe-reports REPORT_DATE OUTPUT_DIRECTORY [--log-level=LEVEL]
+  pe-reports REPORT_DATE OUTPUT_DIRECTORY [--log-level=LEVEL] [--soc_med_included]
```
```suggestion
  pe-reports REPORT_DATE OUTPUT_DIRECTORY [--log-level=LEVEL] [--soc-med-included]
```

jsf9k commented on src/pe_reports/report_generator.py:
```diff
@@ -11,6 +11,7 @@
   -l --log-level=LEVEL              If specified, then the log level will be set to
                                     the specified value.  Valid values are "debug", "info",
                                     "warning", "error", and "critical". [default: info]
+  -s --soc_med_included             Include social media posts from Cybersixgill in the report.
```
```suggestion
  -s --soc-med-included             Include social media posts from Cybersixgill in the report.
```

jsf9k reviewed (changes requested):
The other CLI options use dashes, not underscores.  Can this one do so as well?

I'm having deja vu about this, so apologies if I already requested it.  If so then just remind me why it can't be done.

DJensen94 committed:
replace underscores with dashes

replace underscores with dashes in docopt parameter

Co-authored-by: Shane Frasier <jeremy.frasier@gwe.cisa.dhs.gov>

DJensen94 committed:
replace underscores with dashes

replace underscores with dashes in docopt parameter

Co-authored-by: Shane Frasier <jeremy.frasier@gwe.cisa.dhs.gov>

jsf9k commented on src/pe_reports/report_generator.py:
```diff
@@ -265,12 +260,11 @@ def main():
+    generate_reports(
         validated_args["REPORT_DATE"],
         validated_args["OUTPUT_DIRECTORY"],
+        validated_args["--soc_med_included"],
```
This instance of `--soc_med_included` needs to be updated as well:
```suggestion
        validated_args["--soc-med-included"],
```

DJensen94 committed:
replace underscores with dashes

replace underscores with dashes in docopt parameters

Co-authored-by: Shane Frasier <jeremy.frasier@gwe.cisa.dhs.gov>
